src/commands/e2e/TestRunner.ts

```diff
@@ -1,37 +1,48 @@
 import * as path from 'path';
 import * as fs from 'fs';
 import {WdioConfigGenerator} from './WdioConfigGenerator';
+import {getPathToE2E, getPathToSpecFiles} from './util';
+import * as glob from 'glob';
 
 export class TestRunner {
 
     private readonly plugins: string[];
     private readonly workingDir: string;
     private readonly mainRepoDir: string;
+    private readonly specsParameter: string;
 
     constructor(plugins: string[],
-                workingDir: string, mainRepoDir: string) {
+                workingDir: string, mainRepoDir: string, specsParameter: string) {
         this.workingDir = workingDir;
         this.plugins = plugins;
         this.mainRepoDir = mainRepoDir;
+        this.specsParameter = specsParameter;
     }
 
     public async runTests(): Promise<void> {
         const launcherModule = require(this.getWdioCliExecutable());
         let hasFailedTest: boolean = false;
         for (const plugin of this.plugins) {
-            const wdioConf = path.join(this.workingDir, plugin, 'assets', 'e2e', WdioConfigGenerator.WDIO_CONF_NAME);
-            const launcher = new launcherModule.default(wdioConf, {args: ['']});
-            const testHasFailed: boolean = await launcher.run();
-            if (testHasFailed) {
-                console.warn(`One of the E2E Tests in plugin ${plugin} has failed`);
-                hasFailedTest = true;
+            if (this.specsParameter && this.isSpecInPlugin(plugin) || !this.specsParameter) {
```

Comment:
Please add `( )` to explicitly state order of evaluation of the `&&` and `||` expressions :)